src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -102,6 +104,23 @@ internal SdkDirectoryWorkloadManifestProvider(string sdkRootPath, string sdkVers
                     }
                 }
             }
+            else
```

Comment:
(This is really a comment for line 96, but) GetrWorkloadVersionFromGlobalJson already seems to check if string.IsNullOrEmpty(globalJsonPath) and just returns null in that case, so we should be able to skip the outer if.

Reply:
Good catch!  This was actually a bug where if you had a global.json that didn't specify a workload version, it would ignore the install state.